engine-spring/core/src/main/java/org/camunda/bpm/engine/spring/SpringBeansResolverFactory.java

```diff
@@ -65,7 +71,19 @@ public boolean containsKey(Object key) {
   @Override
   public Object get(Object key) {
     if (key instanceof String) {
-      return applicationContext.getBean((String) key);
+      try {
+        return applicationContext.getBean((String) key);
+      } catch (BeanCreationException ex) {
+        if (SCOPE_NOT_ACTIVE.equals(ex.getClass().getName())) {
```

Comment:
Only swallow exceptions for not beans with inactive scope. Unfortunately, we cannot use `ScopeNotActiveException` directly since it is only available since Spring 5.3, but we still support Spring 4.